pkg/manifests/manifests.go

```diff
@@ -79,17 +79,15 @@ func (f *Factory) RouterDaemonSet(cr *ingressv1alpha1.ClusterIngress) (*appsv1.D
 
 	ds.Name = name
 
-	ds.Spec.Template.Labels = map[string]string{
-		"app":    "router",
-		"router": name,
+	if ds.Spec.Template.Labels == nil {
+		ds.Spec.Template.Labels = map[string]string{}
 	}
+	ds.Spec.Template.Labels["router"] = name
```

Comment:
We should update `assets/router/daemonset.yaml` to indicate that only the one field, and not the entirety of `.spec.template.metadata.labels`, is replaced at uptime.

Reply:
righto